cluster.tf

```diff
@@ -49,7 +49,7 @@ resource "aws_security_group_rule" "cluster_egress_internet" {
   count             = var.cluster_create_security_group ? 1 : 0
   description       = "Allow cluster egress access to the Internet."
   protocol          = "-1"
-  security_group_id = aws_security_group.cluster[0].id
+  security_group_id = join("", aws_security_group.cluster.*.id)
```

Comment:
Because that would change the logic. If you are saying that it would be good to use the local value there I have no objection. I actually think that makes more sense.
I can easily change that, let me know.

Reply:
Are we really changing the logic ? Let's take an example ...

```hcl
locals {
  cluster_security_group_id = var.cluster_create_security_group ? aws_security_group.cluster[0].id : var.cluster_security_group_id
}

resource "aws_security_group_rule" "cluster_egress_internet_1" {
  count             = var.cluster_create_security_group ? 1 : 0
  description       = "Allow cluster egress access to the Internet."
  protocol          = "-1"
  security_group_id = aws_security_group.cluster[0].id
  cidr_blocks       = ["0.0.0.0/0"]
  from_port         = 0
  to_port           = 0
  type              = "egress"
}

resource "aws_security_group_rule" "cluster_egress_internet_2" {
  count             = var.cluster_create_security_group ? 1 : 0
  description       = "Allow cluster egress access to the Internet."
  protocol          = "-1"
  security_group_id = local.cluster_security_group_id
  cidr_blocks       = ["0.0.0.0/0"]
  from_port         = 0
  to_port           = 0
  type              = "egress"
}
```

Will these `aws_security_group_rule` create different security group rules ?

- If `var.cluster_create_security_group` is true, then Terraform will create a rule with `aws_security_group.cluster[0].id` => `local.cluster_security_group_id == aws_security_group.cluster[0].id`
- If not, Terraform won't create any rule at all and local.cluster_security_group_id == var.cluster_security_group_id` 

I'm maybe doing something wrong, but I can't figure out how those SG rules are different.

What do you think ?

Reply:
I agree! I see what you mean now. I missed the fact that the rule resource is conditional on the same variable that determines the value of the local.

The reason why it already made more sense to me to just use the locals is that I thought that even if the security group was passed into the module, the module would still create the rules. 

I'll make those changes. Thanks